src/json/builder.cr

```diff
@@ -94,35 +94,55 @@ class JSON::Builder
   # This method can also be used to write the name of an object field.
   def string(value)
     string = value.to_s
+
     scalar(string: true) do
       io << '"'
-      string.each_char do |char|
-        case char
+
+      pos = 0
+      reader = Char::Reader.new(string)
+
+      while reader.has_next?
+        set_pos = true
+        escape = nil
+
+        case char = reader.current_char
         when '\\'
-          io << "\\\\"
+          escape = "\\\\"
         when '"'
-          io << "\\\""
+          escape = "\\\""
         when '\b'
-          io << "\\b"
+          escape = "\\b"
         when '\f'
-          io << "\\f"
+          escape = "\\f"
         when '\n'
-          io << "\\n"
+          escape = "\\n"
         when '\r'
-          io << "\\r"
+          escape = "\\r"
         when '\t'
-          io << "\\t"
+          escape = "\\t"
         when .ascii_control?
+          io.write string.to_slice[pos, reader.pos - pos]
           io << "\\u"
           ord = char.ord
           io << '0' if ord < 0x1000
           io << '0' if ord < 0x100
           io << '0' if ord < 0x10
           ord.to_s(16, io)
         else
-          io << char
+          set_pos = false
```

Comment:
You could just insert `next` here, eliminating the need for a variable `set_pos`.

Reply:
maybe, ends up being same longer

Reply:
The code might be a bit longer, but I expect execution time to be lower.

Reply:
i changed code, same time. so stay the same for me.

Reply:
I would like to remove `set_pos` personally... it's not a big deal for me but the logic is slightly harder to follow.